src/gax.ts

```diff
@@ -292,23 +342,148 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (options) {
```

Comment:
style: since most of the function logic is inside `if (options)`, it might be more readable code (and less indentation) if you handle `if (!options)` first, return early, and then handle the main case.  go/tott/680

Reply:
lol i just saw this tott today. you got it!